Build failure: swiftPackages.swift-unwrapped

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

```
nix shell "github:NixOS/nixpkgs/nixos-unstable#swiftPackages.swift-unwrapped"
```

### Can Hydra reproduce this build failure?

Hydra is currently rebuilding this package.

### Link to Hydra build job

https://hydra.nixos.org/build/313262126

### Relevant log output

```console
swift> [1181/1339] Linking CXX executable bin/swift-reflection-dump
swift> [1182/1339] Building CXX object tools/SourceKit/lib/SwiftLang/CMakeFiles/SourceKitSwiftLang.dir/SwiftSourceDocInfo.cpp.o
swift> In file included from /build/src/swift/tools/SourceKit/lib/SwiftLang/SwiftSourceDocInfo.cpp:19:
swift> In file included from /build/src/swift/tools/SourceKit/lib/SwiftLang/SwiftLangSupport.h:16:
swift> In file included from /build/src/swift/tools/SourceKit/lib/SwiftLang/CodeCompletion.h:18:
swift> In file included from /build/src/swift/include/swift/IDE/CodeCompletionResult.h:17:
swift> In file included from /build/src/swift/include/swift/IDE/CodeCompletionResultType.h:16:
swift> In file included from /build/src/swift/include/swift/AST/Types.h:24:
swift> In file included from /build/src/swift/include/swift/AST/ExtInfo.h:24:
swift> In file included from /build/src/swift/include/swift/AST/ClangModuleLoader.h:19:
swift> In file included from /build/src/llvm-project/clang/include/clang/AST/DeclTemplate.h:17:
swift> In file included from /build/src/llvm-project/clang/include/clang/AST/ASTConcept.h:17:
swift> In file included from /build/src/llvm-project/clang/include/clang/AST/Expr.h:19:
swift> /build/src/llvm-project/clang/include/clang/AST/Decl.h:151:16: warning: virtual method 'anchor' is inside a 'final' class and can never be overridden [-Wunnecessary-virtual-specifier]
swift>   151 |   virtual void anchor();
swift>       |                ^
swift> /build/src/llvm-project/clang/include/clang/AST/Decl.h:185:16: warning: virtual method 'anchor' is inside a 'final' class and can never be overridden [-Wunnecessary-virtual-specifier]
swift>   185 |   virtual void anchor();
swift>       |                ^
swift> /build/src/llvm-project/clang/include/clang/AST/Decl.h:4651:16: warning: virtual method 'anchor' is inside a 'final' class and can never be overridden [-Wunnecessary-virtual-specifier]
swift>  4651 |   virtual void anchor();
swift>       |                ^
swift> In file included from /build/src/swift/tools/SourceKit/lib/SwiftLang/SwiftSourceDocInfo.cpp:19:
swift> In file included from /build/src/swift/tools/SourceKit/lib/SwiftLang/SwiftLangSupport.h:16:
swift> In file included from /build/src/swift/tools/SourceKit/lib/SwiftLang/CodeCompletion.h:18:
swift> In file included from /build/src/swift/include/swift/IDE/CodeCompletionResult.h:17:
swift> In file included from /build/src/swift/include/swift/IDE/CodeCompletionResultType.h:16:
swift> In file included from /build/src/swift/include/swift/AST/Types.h:24:
swift> In file included from /build/src/swift/include/swift/AST/ExtInfo.h:24:
swift> In file included from /build/src/swift/include/swift/AST/ClangModuleLoader.h:19:
swift> In file included from /build/src/llvm-project/clang/include/clang/AST/DeclTemplate.h:21:
swift> /build/src/llvm-project/clang/include/clang/AST/DeclCXX.h:3144:16: warning: virtual method 'anchor' is inside a 'final' class and can never be overridden [-Wunnecessary-virtual-specifier]
swift>  3144 |   virtual void anchor();
swift>       |                ^
swift> In file included from /build/src/swift/tools/SourceKit/lib/SwiftLang/SwiftSourceDocInfo.cpp:19:
swift> In file included from /build/src/swift/tools/SourceKit/lib/SwiftLang/SwiftLangSupport.h:30:
swift> In file included from /build/src/swift/include/swift/IDETool/CompileInstance.h:16:
swift> In file included from /build/src/swift/include/swift/Frontend/Frontend.h:23:
swift> In file included from /build/src/swift/include/swift/AST/IRGenOptions.h:31:
swift> In file included from /build/src/llvm-project/llvm/include/llvm/Transforms/Instrumentation.h:20:
swift> In file included from /build/src/llvm-project/llvm/include/llvm/IR/IRBuilder.h:25:
swift> /build/src/llvm-project/llvm/include/llvm/IR/ConstantFolder.h:31:16: warning: virtual method 'anchor' is inside a 'final' class and can never be overridden [-Wunnecessary-virtual-specifier]
swift>    31 |   virtual void anchor();
swift>       |                ^
swift> In file included from /build/src/swift/tools/SourceKit/lib/SwiftLang/SwiftSourceDocInfo.cpp:19:
swift> In file included from /build/src/swift/tools/SourceKit/lib/SwiftLang/SwiftLangSupport.h:30:
swift> In file included from /build/src/swift/include/swift/IDETool/CompileInstance.h:16:
swift> In file included from /build/src/swift/include/swift/Frontend/Frontend.h:43:
swift> In file included from /build/src/swift/include/swift/Subsystems.h:20:
swift> In file included from /build/src/swift/include/swift/AST/TBDGenRequests.h:24:
swift> In file included from /build/src/swift/include/swift/IRGen/Linking.h:23:
swift> In file included from /build/src/swift/include/swift/SIL/SILFunction.h:26:
swift> In file included from /build/src/swift/include/swift/SIL/SILBasicBlock.h:24:
swift> /build/src/swift/include/swift/SIL/SILInstruction.h:1149:17: warning: 'offsetof' on non-standard-layout type 'SingleValueInstruction' [-Winvalid-offsetof]
swift>  1149 |   static_assert(offsetof(SingleValueInstruction, kind) ==
swift>       |                 ^                                ~~~~
swift> /nix/store/k2s6xfca52n25yb8b6f58i87kszp3m7g-clang-wrapper-21.1.2/resource-root/include/__stddef_offsetof.h:16:24: note: expanded from macro 'offsetof'
swift>    16 | #define offsetof(t, d) __builtin_offsetof(t, d)
swift>       |                        ^                     ~
swift> In file included from /build/src/swift/tools/SourceKit/lib/SwiftLang/SwiftSourceDocInfo.cpp:19:
swift> In file included from /build/src/swift/tools/SourceKit/lib/SwiftLang/SwiftLangSupport.h:30:
swift> In file included from /build/src/swift/include/swift/IDETool/CompileInstance.h:16:
swift> In file included from /build/src/swift/include/swift/Frontend/Frontend.h:43:
swift> In file included from /build/src/swift/include/swift/Subsystems.h:20:
swift> In file included from /build/src/swift/include/swift/AST/TBDGenRequests.h:24:
swift> In file included from /build/src/swift/include/swift/IRGen/Linking.h:23:
swift> In file included from /build/src/swift/include/swift/SIL/SILFunction.h:26:
swift> In file included from /build/src/swift/include/swift/SIL/SILBasicBlock.h:24:
swift> /build/src/swift/include/swift/SIL/SILInstruction.h:1150:17: warning: 'offsetof' on non-standard-layout type 'NonSingleValueInstruction' [-Winvalid-offsetof]
swift>  1150 |                 offsetof(NonSingleValueInstruction, kind),
swift>       |                 ^                                   ~~~~
swift> /nix/store/k2s6xfca52n25yb8b6f58i87kszp3m7g-clang-wrapper-21.1.2/resource-root/include/__stddef_offsetof.h:16:24: note: expanded from macro 'offsetof'
swift>    16 | #define offsetof(t, d) __builtin_offsetof(t, d)
swift>       |                        ^                     ~
swift> 7 warnings generated.
swift> [1183/1339] Linking CXX static library lib/libSourceKitSwiftLang.a
swift> [1184/1339] Generating /build/build/swift/bootstrapping0/lib/swift/linux/Swift.swiftmodule/x86_64-unknown-linux-gnu.swiftmodule
swift> /build/src/swift/stdlib/public/core/SIMDVector.swift:63:17: warning: protocol 'SIMDScalar' should be declared to refine 'Decodable' due to a same-type constraint on 'Self'
swift> public protocol SIMDScalar {
swift>                 ^
swift> /build/src/swift/stdlib/public/core/SIMDVector.swift:63:17: warning: protocol 'SIMDScalar' should be declared to refine 'Encodable' due to a same-type constraint on 'Self'
swift> public protocol SIMDScalar {
swift>                 ^
swift> /build/src/swift/stdlib/public/core/SIMDVector.swift:63:17: warning: protocol 'SIMDScalar' should be declared to refine 'Hashable' due to a same-type constraint on 'Self'
swift> public protocol SIMDScalar {
swift>                 ^
swift> [1185/1339] Compiling /build/build/swift/bootstrapping0/stdlib/public/core/LINUX/x86_64/Swift.o
swift> /build/src/swift/stdlib/public/core/SIMDVector.swift:63:17: warning: protocol 'SIMDScalar' should be declared to refine 'Decodable' due to a same-type constraint on 'Self'
swift> public protocol SIMDScalar {
swift>                 ^
swift> /build/src/swift/stdlib/public/core/SIMDVector.swift:63:17: warning: protocol 'SIMDScalar' should be declared to refine 'Encodable' due to a same-type constraint on 'Self'
swift> public protocol SIMDScalar {
swift>                 ^
swift> /build/src/swift/stdlib/public/core/SIMDVector.swift:63:17: warning: protocol 'SIMDScalar' should be declared to refine 'Hashable' due to a same-type constraint on 'Self'
swift> public protocol SIMDScalar {
swift>                 ^
swift> [1186/1339] Building C object stdlib/public/core/CMakeFiles/swiftCore-bootstrapping0.dir/__/__/linker-support/magic-symbols-for-install-name.c.o
swift> [1187/1339] Linking CXX shared library bootstrapping0/lib/swift/linux/libswiftCore.so
swift> clang-16-unwrapped: warning: argument unused during compilation: '-nostdlibinc' [-Wunused-command-line-argument]
swift> [1188/1339] Building swift module Basic
swift> FAILED: bootstrapping1/SwiftCompilerSources/Basic.o /build/build/swift/bootstrapping1/SwiftCompilerSources/Basic.o 
swift> cd /build/src/swift/SwiftCompilerSources && /build/build/swift/bootstrapping0/bin/swiftc -c -o /build/build/swift/bootstrapping1/SwiftCompilerSources/Basic.o -I /build/build/swift/bootstrapping0/bin/../lib -I /usr/lib -target x86_64-unknown-linux-gnu -module-name Basic -emit-module -emit-module-path /build/build/swift/bootstrapping1/SwiftCompilerSources/Basic.swiftmodule -parse-as-library /build/src/swift/SwiftCompilerSources/Sources/Basic/SourceLoc.swift /build/src/swift/SwiftCompilerSources/Sources/Basic/Utils.swift -wmo -Xfrontend -validate-tbd-against-ir=none -Xfrontend -enable-experimental-cxx-interop -Xcc -std=c++17 -Xcc -UIBOutlet -Xcc -UIBAction -Xcc -UIBInspectable -Xfrontend -disable-implicit-string-processing-module-import -O -cross-module-optimization -Xcc -DNDEBUG -Xcc -I -Xcc /build/src/llvm-project/llvm/include -Xcc -I -Xcc /build/build/llvm/include -Xcc -I -Xcc /build/src/llvm-project/clang/include -Xcc -I -Xcc /build/build/llvm/tools/clang/include -Xcc -I -Xcc /build/src/swift/include -Xcc -I -Xcc /build/build/swift/SwiftCompilerSources/../include -I /build/build/swift/bootstrapping1/SwiftCompilerSources
swift> <unknown>:0: warning: the '-enable-experimental-cxx-interop' flag is deprecated; please pass '-cxx-interoperability-mode=' instead
swift> <unknown>:0: note: Swift will maintain source compatibility for imported APIs based on the selected compatibility mode, so updating the Swift compiler will not change how APIs are imported
swift> <unknown>:0: warning: libstdc++ not found for 'x86_64-unknown-linux-gnu'; C++ stdlib may be unavailable
swift> <unknown>:0: warning: unable to perform implicit import of "_Concurrency" module: no such module found
swift> /build/src/llvm-project/llvm/include/llvm/CAS/CASReference.h:12:2: note: in module 'LLVM_Utils' imported from /build/src/llvm-project/llvm/include/llvm/CAS/CASReference.h:12:
swift> #include "llvm/ADT/ArrayRef.h"
swift>  ^
swift> /build/src/llvm-project/llvm/include/llvm/Support/type_traits.h:35:13: error: missing '#include <type_traits>'; 'is_class' must be defined before it is used
swift>       !std::is_class<UnderlyingT>::value && // Filter conversion operators.
swift>             ^
swift> /build/src/llvm-project/llvm/include/llvm/ADT/Hashing.h:351:36: note: in instantiation of template class 'llvm::is_integral_or_enum<const void *>' requested here
swift>   : std::integral_constant<bool, ((is_integral_or_enum<T>::value ||
swift>                                    ^
swift> /build/src/llvm-project/llvm/include/llvm/ADT/Hashing.h:368:18: note: in instantiation of template class 'llvm::hashing::detail::is_hashable_data<const void *>' requested here
swift> std::enable_if_t<is_hashable_data<T>::value, T>
swift>                  ^
swift> /build/src/llvm-project/llvm/include/llvm/ADT/Hashing.h:560:63: note: while substituting deduced template arguments into function template 'get_hashable_data' [with T = const void *]
swift>     buffer_ptr = combine_data(length, buffer_ptr, buffer_end, get_hashable_data(arg));
swift>                                                               ^
swift> /build/src/llvm-project/llvm/include/llvm/ADT/Hashing.h:608:17: note: in instantiation of function template specialization 'llvm::hashing::detail::hash_combine_recursive_helper::combine<const void *, const void *>' requested here
swift>   return helper.combine(0, helper.buffer, helper.buffer + 64, args...);
swift>                 ^
swift> /build/src/swift/include/swift/Basic/SourceLoc.h:291:12: note: in instantiation of function template specialization 'llvm::hash_combine<const void *, const void *>' requested here
swift>     return hash_combine(Val.Start.getOpaquePointerValue(),
swift>            ^
swift> /nix/store/68ndh04pl2hhhizsarvzwa9cnlp7zj3d-gcc-14.3.0/include/c++/14.3.0/type_traits:635:12: note: definition here is not reachable
swift>     struct is_class
swift>            ^
swift> /build/src/swift/SwiftCompilerSources/Sources/Basic/SourceLoc.swift:13:8: error: could not build C module 'BasicBridging'
swift> import BasicBridging
swift>        ^
swift> ninja: build stopped: subcommand failed.
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.7-cachyos, NixOS, 25.11 (Xantusia), 25.11.20251108.b6a8526`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Determinate Nix 3.12.2) 2.32.2`
 - channels(root): `""`
 - nixpkgs: `/nix/store/ipyw6f3ah05zxy1imaydg5bmr8qmsiyc-source`


### Notify maintainers

@dduan @stephank @trepetti @Trundle

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/compilers/swift/compiler/default.nix
+++ b/pkgs/development/compilers/swift/compiler/default.nix
@@ -348,6 +348,17 @@ stdenv.mkDerivation {
     patch -p1 -d swift -i $TMPDIR/swift-separate-lib.patch
 
     patch -p1 -d llvm-project/llvm -i ${./patches/llvm-module-cache.patch}
+    for root in llvm-project/llvm swift/stdlib; do
+      patch -p1 -d $root -i ${
+        (fetchpatch {
+          name = "fix-SmallVector-compile-error.patch";
+          url = "https://github.com/llvm/llvm-project/commit/7e44305041d96b064c197216b931ae3917a34ac1.patch";
+          stripLen = 1;
+          hash = "sha256-1htuzsaPHbYgravGc1vrR8sqpQ/NSQ8PUZeAU8ucCFk=";
+        })
+      }
+    done
+    patch -p2 -d llvm-project/llvm -i ${./patches/llvm-fix-X86MCTargetDesc-compile-error.patch}
 
     for lldbPatch in ${
       lib.escapeShellArgs [
@@ -362,6 +373,12 @@ stdenv.mkDerivation {
           url = "https://github.com/llvm/llvm-project/commit/68744ffbdd7daac41da274eef9ac0d191e11c16d.patch";
           stripLen = 1;
           hash = "sha256-QCGhsL/mi7610ZNb5SqxjRGjwJeK2rwtsFVGeG3PUGc=";
+        })
+        (fetchpatch {
+          name = "LLDB-Add-cstdint-to-AddressableBits-102110.patch";
+          url = "https://github.com/llvm/llvm-project/commit/bb59f04e7e75dcbe39f1bf952304a157f0035314.patch";
+          stripLen = 1;
+          hash = "sha256-+CcmZRxCaozFe1Kuf2HX+kGKuh/PDuoFBEFA/t7tL9A=";
         })
       ]
     }; do

--- a/pkgs/development/compilers/swift/compiler/patches/llvm-fix-X86MCTargetDesc-compile-error.patch
+++ b/pkgs/development/compilers/swift/compiler/patches/llvm-fix-X86MCTargetDesc-compile-error.patch
@@ -1,0 +1,34 @@
+From f03ab75680385b1ea62af3ab15c423a3bc0f3588 Mon Sep 17 00:00:00 2001
+From: Stephan Hageboeck <stephan.hageboeck@cern.ch>
+Date: Mon, 20 Jan 2025 17:52:47 +0100
+Subject: [PATCH] Add missing include to X86MCTargetDesc.h (#123320)
+
+In gcc-15, explicit includes of `<cstdint>` are required when fixed-size
+integers are used. In this file, this include only happened as a side
+effect of including SmallVector.h
+
+Although llvm compiles fine, the root-project would benefit from
+explicitly including it here, so we can backport the patch.
+
+Maybe interesting for @hahnjo and @vgvassilev
+
+(cherry picked from commit 7abf44069aec61eee147ca67a6333fc34583b524)
+---
+ llvm/lib/Target/X86/MCTargetDesc/X86MCTargetDesc.h | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/llvm/lib/Target/X86/MCTargetDesc/X86MCTargetDesc.h b/llvm/lib/Target/X86/MCTargetDesc/X86MCTargetDesc.h
+index d0530bd4d650..10b59462aebe 100644
+--- a/llvm/lib/Target/X86/MCTargetDesc/X86MCTargetDesc.h
++++ b/llvm/lib/Target/X86/MCTargetDesc/X86MCTargetDesc.h
+@@ -13,6 +13,7 @@
+ #ifndef LLVM_LIB_TARGET_X86_MCTARGETDESC_X86MCTARGETDESC_H
+ #define LLVM_LIB_TARGET_X86_MCTARGETDESC_X86MCTARGETDESC_H
+ 
++#include <cstdint>
+ #include <memory>
+ #include <string>
+ 
+-- 
+2.52.0
+

--- a/pkgs/development/compilers/swift/compiler/patches/swift-linux-fix-libc-paths.patch
+++ b/pkgs/development/compilers/swift/compiler/patches/swift-linux-fix-libc-paths.patch
@@ -21,10 +21,10 @@ patch it to also consider `-idirafter` and `-isystem` as added by cc-wrapper.
 +  llvm::append_range(includeDirs,
 +                     DriverArgs.getAllArgValues(
 +                         clang::driver::options::OPT_idirafter));
-+  // Nix adds the C++ stdlib include path using `-isystem`.
++  // Nix adds the C++ stdlib include path using `-cxx-isystem`.
 +  llvm::append_range(includeDirs,
 +                     DriverArgs.getAllArgValues(
-+                         clang::driver::options::OPT_isystem));
++                         clang::driver::options::OPT_cxx_isystem));
  
    for (const auto &includeDir : includeDirs) {
      Path dir(includeDir);

--- a/pkgs/development/compilers/swift/swiftpm/default.nix
+++ b/pkgs/development/compilers/swift/swiftpm/default.nix
@@ -259,6 +259,7 @@ let
 
     patches = [
       ./patches/llbuild-cmake-disable-rpath.patch
+      ./patches/llbuild-fix-missing-cstdint.patch
     ];
 
     postPatch = ''

--- a/pkgs/development/compilers/swift/swiftpm/patches/llbuild-fix-missing-cstdint.patch
+++ b/pkgs/development/compilers/swift/swiftpm/patches/llbuild-fix-missing-cstdint.patch
@@ -1,0 +1,12 @@
+diff --git a/utils/unittest/googletest/src/gtest-death-test.cc b/utils/unittest/googletest/src/gtest-death-test.cc
+index ede8378..bdef225 100644
+--- a/utils/unittest/googletest/src/gtest-death-test.cc
++++ b/utils/unittest/googletest/src/gtest-death-test.cc
+@@ -33,6 +33,7 @@
+ #include "gtest/gtest-death-test.h"
+ 
+ #include <utility>
++#include <cstdint>
+ 
+ #include "gtest/internal/gtest-port.h"
+ #include "gtest/internal/custom/gtest.h"

--- a/pkgs/development/compilers/swift/wrapper/wrapper.sh
+++ b/pkgs/development/compilers/swift/wrapper/wrapper.sh
@@ -188,7 +188,7 @@ addCFlagsToList() {
             # Pass through using -Xcc, but also convert to Swift -I.
             # These have slightly different meaning for Clang, but Swift
             # doesn't have exact equivalents.
-            -isystem | -idirafter)
+            -isystem | -cxx-isystem | -idirafter)
                 i=$((i + 1))
                 list+=("-Xcc" "$val" "-Xcc" "${!i}" "-I" "${!i}")
                 ;;

